app/controllers/admin/categories_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Admin::CategoriesController < ApplicationController
+
+  def new
```

Comment:
Please add the before_action here which will restrict the normal users.

before_action :logged_in_user
before_action :admin_user